### DISCUSSION TITLE
Is it possible to use DRAM to load GPU_IVF_FLAT index?

### DISCUSSION THREAD

**Discussion by RakeshRaj97** [milvus-io/milvus#34016] [Q&A and General discussion]
I have a very large collection of size 10 billion entries. I would like to use GPUs to index it faster and I want to make use of DRAM to load the collection and search it using CPUs. Is this possible given my bottleneck is the GPU VRAM and its unable to load the collection?

**Comment by yhmo** [comment]
Swap data between CPU memory and GPU memory? Even milvus allows to do that, the search performance will be poor.

**Reply by RakeshRaj97** [reply]
I mean load them to a CPU only node and do a search only using CPUs. I have multiple CPU only nodes which has 500G DRAM capacity and that can hold the entire 10 billion vectors at once.

**Comment by xiaofan-luan** [comment]
how large the NQ is?
is this a offline use case?

**Reply by RakeshRaj97** [reply]
The NQ is 500 and it's an offline use case.

**Comment by xiaofan-luan** [comment]
This might be reasonable use case for offline use case.

all data is on local disk or memory. if search nq is large, pin one segment in memory, do a batch search and swap to another segment

**Reply by RakeshRaj97** [reply]
You mean CPU memory or GPU memory?  Is there any examples which I can follow to load a segment?
